metricflow/sql/render/big_query.py

```diff
@@ -120,14 +123,18 @@ def visit_cast_to_timestamp_expr(self, node: SqlCastToTimestampExpression) -> Sq
     @override
     def visit_date_trunc_expr(self, node: SqlDateTruncExpression) -> SqlExpressionRenderResult:
         """Render DATE_TRUNC for BigQuery, which takes the opposite argument order from Snowflake and Redshift."""
+        self._validate_granularity_for_engine(node.time_granularity)
+
         arg_rendered = self.render_sql_expr(node.arg)
 
         prefix = ""
         if node.time_granularity == TimeGranularity.WEEK:
             prefix = "iso"
 
+        trunc_expr = "DATE_TRUNC" if node.time_granularity.to_int() >= TimeGranularity.DAY.to_int() else "TIME_TRUNC"
+
```

Comment:
We should use DATETIME_TRUNC for now, and eventually we'll need to shift to either DATETIME_TRUNC or TIMESTAMP_TRUNC.

The good news is we don't need this conditional at the moment, since DATETIME_TRUNC  supports time-level granularities.

Worth noting, BigQuery's TIME type is independent of date. https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#time_type
```suggestion
```